packages/relay/src/lib/eth.ts

```diff
@@ -1828,8 +1827,8 @@ export class EthImpl implements Eth {
    * @param requestIdPrefix
    */
   async performCallChecks(call: any): Promise<void> {
-    // The "to" address must always be 42 chars.
-    if (!call.to || call.to.length != 42) {
+    // after this mirror-node PR https://github.com/hashgraph/hedera-mirror-node/pull/8100, call.to is allowed to be empty or null
+    if (call.to !== null && !isValidEthereumAddress(call.to)) {
```

Comment:
Is the first check necessary.
`isValidEthereumAddress()` should be able to handle `null` or `undefined`. Check if there are tests for this and if not we should add them to confirm.
```suggestion
    if (!isValidEthereumAddress(call.to)) {
```

Reply:
I understand your concern. This situation requires both checks to function correctly. Given that `call.to` can be null or undefined, the `if statement` should not throw an error if `call.to` is null or undefined. 

The function `isValidEthereumAddress(call.to)` returns `false` when `call.to` is null or undefined. Therefore, `!isValidEthereumAddress(call.to)` evaluates to `true`, making the `if statement` valid and potentially causing an error, which is incorrect.

With both checks in place:
- If `call.to` is null or undefined, the error is skipped.
- If `call.to` is not null or undefined, its validity as an Ethereum address is checked.
  - If `call.to` is a valid Ethereum address, the process continues.
  - If `call.to` is not a valid Ethereum address, an error is thrown.